### PR TITLE
Adds pagination to messages recieved

### LINKED ISSUE #43: Implement Pagination in Endpoint for Fetching All Messages
Currently, the endpoint for retrieving all messages returns all message documents from mongoDB as is.  
This is certainly will affect the load time. In order to fix this, we'll implement pagination in that endpoint.  
For starting point, you can search for pagination in mongoose because we use mongoose as ODM for the MongoDB
Route for the endpoint is `GET /message`, you can find it in `route.ts`

### PR DESCRIPTION
**Pull request by TakoLeakest**
- Modifies "/messages" endpoint to recieve "lastId" and "limit" parameters
- "limit" determines how many messages will be recieved after "lastId"
- "lastId" corresponds to the "_id" field of the last message recieved

<br />

- "lastId" is equal to "NULL" by default
- "limit" is equal to 10 by default
- Closes #43 

*Results:*
image

### CODE REVIEW

**Comment by RedMaps** [comment]
Have you considered using an `offset` parameter instead of `lastId`? Seems easier to implement for front and backend

**Comment by TakoLeakest** [comment]
I was considering the use of the "skip" function to, well, skip the first N messages. But [the documentation](https://docs.mongodb.com/manual/reference/method/cursor.skip/#behavior) also mentions being careful with the size when using skip.

image
*Source: https://www.codementor.io/@arpitbhayani/why-mongodb-s-cursor-skip-is-slow-90j5f3fst*

**Comment by TakoLeakest** [comment]
Of course, this is probably for REALLY large offsets, but I prefer to be careful with backend stuff

**Comment by azuraga** [comment]
I agree that `lastId` is better in that case. Unless we find a way to assure that using `skip` is going efficiently.

**Comment by nyxordinal** [comment]
One question
If we want to get the first `n` document then we just need to set the `limit` parameter and ignore `lastId` ?

**Comment by TakoLeakest** [comment]
Yes, ignoring the lastId (Or sending "NULL") returns the first [limit] messages

File: packages/backend/src/controllers/message.ts [hunk 1]
```diff
@@ -8,9 +8,12 @@ import {
 } from '@util/response'
 import { Request, Response } from 'express'
 
-export const getAllMessagesController = async (_: Request, res: Response) => {
+export const getAllMessagesController = async (req: Request, res: Response) => {
   try {
-    const messages: MessageDoc[] = await getAllMessages()
+    // "/messages?lastId=[lastId]&limit=[limit]"
```
Comment (nyxordinal) [inline packages/backend/src/controllers/message.ts]: you can remove this comment and instead include it in the description of this PR

Comment (nyxordinal) [reply packages/backend/src/controllers/message.ts]: We'll include that information in API documentation later

**Comment by nyxordinal** [comment]
Looks good to me, but make sure you've formatted the code

File: packages/backend/src/controllers/message.ts (hunk 1 above)
Comment (TakoLeakest) [reply packages/backend/src/controllers/message.ts]: Ok, I'll do that ina second

**Review by nyxordinal — APPROVED** [review]

**Comment by nyxordinal** [comment]
Good job ! I'll merge this now. Thanks for your contribution

**Pull request merged**